terraform/modules/cloud/aws/gateway/main.tf

```diff
@@ -1,3 +1,74 @@
+resource "aws_acm_certificate" "cert" {
+  domain_name       = "${var.api_dns}"
+  validation_method = "DNS"
+}
+
+resource "aws_route53_record" "cert_validation" {
+  name    = "${aws_acm_certificate.cert.domain_validation_options.0.resource_record_name}"
+  type    = "${aws_acm_certificate.cert.domain_validation_options.0.resource_record_type}"
+  zone_id = "${var.dns_zone}"
+  records = ["${aws_acm_certificate.cert.domain_validation_options.0.resource_record_value}"]
+  ttl     = 60
+}
+
+resource "aws_route53_record" "main-api" {
+  zone_id = "${var.dns_zone}"
+  name    = "${var.api_dns}"
+  type    = "A"
+
+  alias {
+    name                   = "${aws_cloudfront_distribution.cf.domain_name}"
+    zone_id                = "${aws_cloudfront_distribution.cf.hosted_zone_id}"
+    evaluate_target_health = false
+  }
+}
+
+resource "aws_cloudfront_distribution" "cf" {
+  enabled = true
+
+  origin {
+    domain_name = "origin-${var.api_dns}"
+    origin_id   = "main"
+
+    custom_origin_config {
+      http_port              = 80
+      https_port             = 443
+      origin_protocol_policy = "match-viewer"
+      origin_ssl_protocols   = ["TLSv1.2", "SSLv3", "TLSv1", "TLSv1.1"]
+    }
+  }
+
+  default_cache_behavior {
+    allowed_methods = ["DELETE", "GET", "HEAD", "OPTIONS", "PATCH", "POST", "PUT"]
+    cached_methods  = ["GET", "HEAD"]
+
+    forwarded_values {
+      query_string = false
```

Comment:
This is needed as the API uses query params: https://github.com/aeternity/aeternity/blob/master/config/swagger.yaml#L806